src/main/java/fi/aalto/cs/apluscourses/presentation/MainViewModel.java

```diff
@@ -61,7 +62,9 @@ private void updateExercises() {
         .orElse(null);
     Authentication auth = authentication.get();
     if (course == null || auth == null) {
-      exercisesViewModel.set(null);
+      ExercisesTreeViewModel emptyModel = new ExercisesTreeViewModel(new ArrayList<>(),
+              new Options());
+      exercisesViewModel.set(emptyModel);
```

Comment:
Hmm, in another comment I noted that this could be an alternative to the current approach of passing `MainViewModel` to the ctor, since `viewModelChanged` would then always receive a non-null parameter.

But with the current approach, I think this change isn't needed anymore, since I think there's no problem if `exercisesViewModel` is null instead, as it used to be